action.yml

```diff
@@ -2,6 +2,11 @@ name: mergeabot
 description: Auto-merge Dependabot PRs, only after quarantine days
 
 inputs:
+  exclude-regex:
+    description: Exclude PRs whose titles match this regular expression
```

Comment:
The Dependabot PR titles are well-structured and stable, AFAICT.

> Bump {dep} to {version} in {path}

So `exclude-regex: "in /important/path$"` was exactly to accomplish directory-based.

I assume you're suggesting to actually have the action pull down `pull/:n/files` and check it against an `excluded-directories` value.

That is both fewer features (can't exclude by dependency or version) and more complexity (another API call, a loop, and array-inclusion checks in Bash are not fun -- vs a single `$title =~`). If the titles weren't so structured, I'd agree with you that it is too fragile. But given that they are so structured, I chose this implementation and still prefer it.

Reply:
The fragility I'm pointing at isn't so much that the titles are well structured, but that we cannot guarantee their structure. They could easily change because a reasonable developer on the dependabot end would not expect titles to be consequential structure.